python/packages/nisar/workflows/stage_watermask.py

```diff
@@ -367,18 +383,101 @@ def download_watermask(polys, epsgs, outfile, version):
             translate_watermask(vrt_filename, outpath, xmin, xmax, ymin, ymax)
 
         # Build vrt with downloaded watermasks
-        gdal.BuildVRT(outfile, watermask_list)
-    except Exception:
+        vrt_dataset = gdal.BuildVRT(outfile, watermask_list)
+
+        # Get the WATER description from the LICENSE.txt file using GDAL
+        in_license_path = vrt_filename.replace(f'EPSG{epsg}.vrt',
+                                               f'EPSG{epsg}/LICENSE.txt')
+        license_exists = gdal.VSIStatL(in_license_path) is not None
+
+        if license_exists:
+            try:
+                water_descr = parse_license_notes(in_license_path)
+                vrt_dataset.SetMetadataItem("water_mask_description",
+                                            f'{water_descr}')
```

Comment:
It should already be a string, so no need for this

```suggestion
                                            water_descr)
```